### PR TITLE
DB-8422 Security Audit Log (create/drop user, modify password, login)

### PR DESCRIPTION
**Pull request by changli6**
This has the same content as the previous reverted PR.

### CODE REVIEW

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2
Testing if it caused failures this time.

**Comment by spliceci** [comment]
TEST FAILED

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by jyuanca** [comment]
+1 except some minor issues

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java [hunk 1]
```diff
@@ -68,6 +68,7 @@
 import com.splicemachine.db.impl.sql.compile.CompilerContextImpl;
 import com.splicemachine.db.impl.sql.execute.*;
 import com.splicemachine.db.impl.sql.misc.CommentStripper;
+import com.splicemachine.utils.StringUtils;
 import org.apache.log4j.Level;
```
[the remark this answers is not in the document]
Comment (changli6) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java]: Removed

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericStatementContext.java [hunk 2]
```diff
@@ -60,6 +60,7 @@
 
 import com.splicemachine.db.iapi.error.ExceptionSeverity;
 import com.splicemachine.db.iapi.reference.SQLState;
+import com.splicemachine.utils.StringUtils;
 import org.apache.log4j.Logger;
```
[the remark this answers is not in the document]
Comment (changli6) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericStatementContext.java]: Removed

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**